pulser/simulation/simconfig.py

```diff
@@ -93,11 +93,15 @@ class SimConfig:
 
     def __post_init__(self) -> None:
         self._process_temperature()
-        self.__dict__["spam_dict"] = {
-            "eta": self.eta,
-            "epsilon": self.epsilon,
-            "epsilon_prime": self.epsilon_prime,
-        }
+        object.__setattr__(
```

Comment:
I'm thinking that on cases where you do this a lot, it might make sense to make a private method that "hides away" the ugliness, something like:
```
def _change_value(self, name, new_value):
    object.__setattr__(self, name, new_value)
```
What do you think?

Reply:
I guess in `simconfig.py` and `variable.py` can make sense. Or would you add it globally?

Reply:
Actually though I'm not sure..It's not like the syntax or the line length changes much

Reply:
It's true, but it prevents other users to have to know this weird thing. I would suggest doing it on the classes that have potential to grow, which would be just `SimConfig`.

Reply:
sounds good, I've made the change let me know if it's okay